drizzle-arktype/src/schema.types.internal.ts

```diff
@@ -0,0 +1,106 @@
+import type { Type, type } from 'arktype';
+import { Inferred } from 'arktype/internal/methods/base.ts';
+import type { Assume, Column, DrizzleTypeError, SelectedFieldsFlat, Simplify, Table, View } from 'drizzle-orm';
+import type {
+	ArktypeNullable,
+	ArktypeOptional,
+	GetArktypeType,
+	GetBaseColumn,
+	GetEnumValuesFromColumn,
+	HandleColumn,
+} from './column.types.ts';
+import type { GetSelection, RemoveNever } from './utils.ts';
+
+export interface Conditions {
+	never: (column?: Column) => boolean;
+	optional: (column: Column) => boolean;
+	nullable: (column: Column) => boolean;
+}
+
+type GenericSchema = Inferred | [Inferred, '?'];
+
+export type BuildRefineColumns<
+	TColumns extends Record<string, any>,
+> = Simplify<
+	RemoveNever<
+		{
+			[K in keyof TColumns]: TColumns[K] extends infer TColumn extends Column ? GetArktypeType<
+					TColumn['_']['data'],
+					TColumn['_']['dataType'],
+					TColumn['_']['columnType'],
+					GetEnumValuesFromColumn<TColumn>,
+					GetBaseColumn<TColumn>
+				> extends infer TSchema extends GenericSchema ? TSchema
+				: Type<any, {}>
+				: TColumns[K] extends infer TObject extends SelectedFieldsFlat<Column> | Table | View
+					? BuildRefineColumns<GetSelection<TObject>>
+				: TColumns[K];
+		}
+	>
+>;
+
+export type BuildRefine<
+	TColumns extends Record<string, any>,
+> = BuildRefineColumns<TColumns> extends infer TBuildColumns ? {
+		[K in keyof TBuildColumns]?: TBuildColumns[K] extends GenericSchema
```

Comment:
If I understand correctly what you're looking to do here, the easiest + most efficient way is `TBuildColumns[K] extends type.Any ? ((schema: TBuildColumns[K]) => type.Any) | type.Any`.

`Inferred` is internal and shouldn't be used here since there are other ways to identify a `Type` (though totally understand how those are hard to find externally).